src/rdkafka_admin.c

```diff
@@ -4801,3 +5058,1551 @@ void rd_kafka_DeleteAcls(rd_kafka_t *rk,
+                rd_kafka_AlterConsumerGroupOffsets_destroy(alter_grpoffsets[i]);
+}
+
+/**
+ * @brief Allocate a new AlterGroup and make a copy of \p src
+ */
+static rd_kafka_AlterConsumerGroupOffsets_t *
+rd_kafka_AlterConsumerGroupOffsets_copy(
+    const rd_kafka_AlterConsumerGroupOffsets_t *src) {
+        return rd_kafka_AlterConsumerGroupOffsets_new(src->group_id,
+                                                      src->partitions);
+}
+
+/**
+ * @brief Send a OffsetCommitRequest to \p rkb with the partitions
+ *        in alter_grpoffsets (AlterConsumerGroupOffsets_t*) using
+ *        \p options.
+ *
+ */
+static rd_kafka_resp_err_t rd_kafka_AlterConsumerGroupOffsetsRequest(
+    rd_kafka_broker_t *rkb,
+    /* (rd_kafka_AlterConsumerGroupOffsets_t*) */
+    const rd_list_t *alter_grpoffsets,
+    rd_kafka_AdminOptions_t *options,
+    char *errstr,
+    size_t errstr_size,
+    rd_kafka_replyq_t replyq,
+    rd_kafka_resp_cb_t *resp_cb,
+    void *opaque) {
+        const rd_kafka_AlterConsumerGroupOffsets_t *grpoffsets =
+            rd_list_elem(alter_grpoffsets, 0);
+
+        rd_assert(rd_list_cnt(alter_grpoffsets) == 1);
+
+        rd_kafka_topic_partition_list_t *offsets = grpoffsets->partitions;
+        rd_kafka_consumer_group_metadata_t *cgmetadata =
+            rd_kafka_consumer_group_metadata_new(grpoffsets->group_id);
+
+        int ret = rd_kafka_OffsetCommitRequest(
+            rkb, cgmetadata, offsets, replyq, resp_cb, opaque,
+            "rd_kafka_AlterConsumerGroupOffsetsRequest");
+        rd_kafka_consumer_group_metadata_destroy(cgmetadata);
+        if (ret == 0) {
+                rd_snprintf(errstr, errstr_size,
+                            "At least one topic-partition offset must "
+                            "be >= 0");
+                return RD_KAFKA_RESP_ERR__NO_OFFSET;
+        }
+        return RD_KAFKA_RESP_ERR_NO_ERROR;
+}
+
+/**
+ * @brief Parse OffsetCommitResponse and create ADMIN_RESULT op.
+ */
+static rd_kafka_resp_err_t
+rd_kafka_AlterConsumerGroupOffsetsResponse_parse(rd_kafka_op_t *rko_req,
+                                                 rd_kafka_op_t **rko_resultp,
+                                                 rd_kafka_buf_t *reply,
+                                                 char *errstr,
+                                                 size_t errstr_size) {
+        rd_kafka_t *rk;
+        rd_kafka_broker_t *rkb;
+        rd_kafka_op_t *rko_result;
+        rd_kafka_topic_partition_list_t *partitions = NULL;
+        rd_kafka_resp_err_t err = RD_KAFKA_RESP_ERR_NO_ERROR;
+        const rd_kafka_AlterConsumerGroupOffsets_t *alter_grpoffsets =
+            rd_list_elem(&rko_req->rko_u.admin_request.args, 0);
+        partitions =
+            rd_kafka_topic_partition_list_copy(alter_grpoffsets->partitions);
+
+        rk  = rko_req->rko_rk;
+        rkb = reply->rkbuf_rkb;
+        err = rd_kafka_handle_OffsetCommit(rk, rkb, err, reply, NULL,
```

Comment:
We are keeping the `request` parameter of this function NULL.
But it's possible that, in case we encounter `RD_KAFKA_RESP_ERR_UNKNOWN_TOPIC_OR_PART`, we will set the action to take as `RD_KAFKA_ERR_ACTION_RETRY`, which will eventually call `rd_kafka_buf_retry(rkb, request)` which will segfault.


Writing this segfault here, I'm not sure how to fix it yet, I'll update the PR with the fix commit when I find one (maybe using ignore_cgrp in some way is the best fix)

Reply:
It's tricky because we can't ignore things like `RD_KAFKA_RESP_ERR_COORDINATOR_LOAD_IN_PROGRESS` in which case we _actually_ want to retry.

Reply:
I'm not sure why it is reusing handle_OffsetCommit() in the first place? That's for another request type,  this should have its own response handler.

Reply:
Ref the KIP 396:
> Regarding the implementation, instead of relying on KafkaConsumer to perform the alter, I propose to instead directly send OffsetCommit requests. This is principally for 2 reasons [...]

We're doing the same thing and inside `rd_kafka_AlterConsumerGroupOffsetsResponse_parse` we call handle_OffsetCommit